### PR TITLE
First design of the OpenShift Pipelines next

### PR DESCRIPTION
**Pull request by chmouel**
I skipped the blah blah and went straight from the bones of the current state to
the meat of the technical implementation details.

Very much interested on your thoughts on goals and requirements, the implementation details
are just well humm details of a mad man 😜,  just to syntethise if we can do it.
 
Probably better to walk thru them  over a VC call.

Signed-off-by: Chmouel Boudjnah <chmouel@chmouel.com>

/cc @vdemeester @sthaha @nikhil-thomas @hrishin

### CODE REVIEW

File: docs/OpenShift-Pipelines-CI.Next-proposal.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+       - checkout `openshift-pipelines/plumbing` repo as tekton git resources.
+
+       - start and wait for an openshift4 temporary cluster install using an
+         openshift ci registry token.
+
+         (**TBD**: which token? a user one or req a stable one?)
+
+        - store generated kubeconfig and password in an artifacts/object storage/content repository
+
+          (**TBD**: we need to figure out a way to store securely store a
```
Comment (chmouel) [inline docs/OpenShift-Pipelines-CI.Next-proposal.md:97]: We could have an emptyDir between tasks for storing secure stuff and store in s3 (if we have access to s3 devint instance) the public yamls artifacts.

Although I very much would love to be able to store to the kubeconfig so we can introspect on a failed run,

File: docs/OpenShift-Pipelines-CI.Next-proposal.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+       -  push it to the internal openshift registry.
+
+   * task2:
+       - run after task1
+
+       - using generated image from previous step.
+
+       - checkout `openshift-pipelines/plumbing` repo as tekton git resources.
+
+       - start and wait for an openshift4 temporary cluster install using an
```
Comment (vdemeester) [inline docs/OpenShift-Pipelines-CI.Next-proposal.md:90]: Can't we do that in parallel of other things ? (like compiling, linting, unit tests, …)

Comment (vdemeester) [reply docs/OpenShift-Pipelines-CI.Next-proposal.md:90]: ah I see, you want to execute everything on the temporary cluster instead of in `cluster-devint`

Comment (chmouel) [reply docs/OpenShift-Pipelines-CI.Next-proposal.md:90]: That's correct, and that's unfortunate :( We want to be able to test against multiple openshift target....

**Comment by chmouel** [comment]
moved to https://hackmd.io/xahO1zddS_m5TzGmhHHdUw?both

**Pull request closed without merging**